src/style/style.js

```diff
@@ -794,6 +794,10 @@ class Style extends Evented {
             this.fire(new ErrorEvent(new Error(`The sourceLayer parameter must be provided for vector source types.`)));
             return;
         }
+        if (feature.id == null) {
```

Comment:
What happens if you pass an undefined value or empty string as the feature id? Is that considered valid?

Reply:
`== null` works for `undefined` and `null`, but not for `""`. Ive added that additional check. Using `===` isn't appropriate here because `0` is a valid id value.

Reply:
👍 good explanation